.github/workflows/publish-pypi.yml

```diff
@@ -0,0 +1,55 @@
+name: pypi publishing
+
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+
+jobs:
+
+  publish_on_pypi:
+
+    if: startsWith(github.ref, 'refs/tags')
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: checkout latest commit
+        uses: actions/checkout@v2
+
+      - name: get history and tags for SCM versioning to work
+        run: |
+          git fetch --prune --unshallow
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          git describe --tags
+          git describe --tags $(git rev-list --tags --max-count=1)
```

Comment:
What exactly do we do it for?

Reply:
I had an issue with the version of the generated packages  (both `sdist` and `bdist_wheel`) that seemed related to `setuptools_scm` being unable to check which version was the right one to use. I also had the following warning
```python
UserWarning: "/home/runner/work/Search/Search" is shallow and may cause errors
0.1.dev1+g767e300
  warnings.warn('"{}" is shallow and may cause errors'.format(wd.path))
```
which led me to this [issue on GitHub](https://github.com/pypa/setuptools_scm/issues/414) where the lines of code you are highlighting now were indicated as the solution. I then tested by triggering the PyPI workflow and it was finally working fine!